core/src/main/java/org/apache/calcite/sql/fun/SqlDatePartFunction.java

```diff
@@ -56,6 +58,11 @@ public SqlDatePartFunction(String name, TimeUnit timeUnit) {
 
   @Override public SqlNode rewriteCall(SqlValidator validator, SqlCall call) {
     final List<SqlNode> operands = call.getOperandList();
+
+    Preconditions.checkArgument(operands.size() == 1,
+        String.format("Invalid number of arguments to function '%s'."
+            + " Was expecting 1 arguments", getName()));
```

Comment:
@asolimando I wrote my version like you suggested. However, I found it was different from the exception message here: https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/runtime/CalciteResource.java#L129 
Then I changed it to keep them aligned. WDYT?

Reply:
It's better to be consistent, you are right, in such a case let's use that exception directly (like [here](https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java#L1932-L1934)).

This will also address the forbiddenapi issue, `String#format()` by default does not respect the LOCALE IIRC.